store.go

```diff
@@ -315,10 +315,10 @@ type Store interface {
 	// ApplyDiffer applies a diff to a layer.
 	// It is the caller responsibility to clean the staging directory if it is not
 	// successfully applied with ApplyDiffFromStagingDirectory.
-	ApplyDiffWithDiffer(to string, options *drivers.ApplyDiffOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error)
+	ApplyDiffWithDiffer(to string, options *drivers.ApplyDiffWithDifferOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error)
 
 	// ApplyDiffFromStagingDirectory uses stagingDirectory to create the diff.
-	ApplyDiffFromStagingDirectory(to, stagingDirectory string, diffOutput *drivers.DriverWithDifferOutput, options *drivers.ApplyDiffOpts) error
+	ApplyDiffFromStagingDirectory(to, stagingDirectory string, diffOutput *drivers.DriverWithDifferOutput, options *drivers.ApplyDiffWithDifferOpts) error
```

Comment:
the reason why I've added a new type is that the existing struct is used as the output, and I wanted to make clear the data that is passed to the function and the data that is returned.

The user is https://github.com/containers/image/pull/1980/commits/cbea9be44d374151757269f4fec431acad059206#diff-2d4a826a6986cd3a8f73b9fb0e68b27e87caaefb0703a5b1695255e4d22a7d5aR673-R678

Reply:
I’m sorry, `ApplyDiffOpts` is actually used in non-TOC `ApplyDiff`; I confused this value with `DriverWithDifferOutput` somehow.

---

Thanks for the pointer,

If the new `Flags` are only set, and only read, when calling `ApplyDiffFromStagingDirectory`, shouldn’t they only supplied to that function, and _not_ also added to `ApplyDiffWithDiffer`?

And, at least right now (?)`ApplyDiffFromStagingDirectory` doesn’t seem to use the existing `ApplyDiffOpts` for anything, so maybe that can be replaced by a new struct which _only_ carries the `Flags`, without having to nest.

Reply:
I don’t have a good intuition about it. My _default_ is that each API call should have its own, specific and individually documented, `…Opts` struct unless there is a specific reason to share.

Maybe there is a good reason to share? At least looking at the one existing implementation so far, I can’t see it; but I have no idea how this might change e.g. with more native composers features.